crypto/dilithium/pqcrystals_dilithium_ref_common/sign.c

```diff
@@ -225,6 +238,26 @@ int crypto_sign_signature_internal(ml_dsa_params *params,
   /* FIPS 204: line 33 Write signature */
   pack_sig(params, sig, sig, &z, &h);
   *siglen = params->bytes;
+
+  /* FIPS 204. Section 3.6.3 Destruction of intermediate values. */
+  OPENSSL_cleanse(seedbuf, sizeof(seedbuf));
+  OPENSSL_cleanse(rho, sizeof(rho));
+  OPENSSL_cleanse(tr, sizeof(tr));
+  OPENSSL_cleanse(key, sizeof(key));
+  OPENSSL_cleanse(mu, sizeof(mu));
```

Comment:
Thank you! I ran through a debugger with full keygen, sign. verify to manually check through the cleansed values. I added correct sizes for the four callouts here, as well as similar cleanses in keygen, and verify.

Since `rho` is of size `SEEDBYTES`, `tr` is of size `TRBYTES`, `mu` is of size `CRHBYTES`, `rhoprime` is of size `CRHBYTES`, `key` is of size `SEEDBYTES` we can use these size macros:

```
  OPENSSL_cleanse(rho, SEEDBYTES);
  OPENSSL_cleanse(tr, TRBYTES);
  OPENSSL_cleanse(key, SEEDBYTES);
  OPENSSL_cleanse(mu, CRHBYTES);
  OPENSSL_cleanse(rhoprime, CRHBYTES);
```